vec3.h

```diff
@@ -0,0 +1,120 @@
+//code by Ameya Vikrama Singh
+#include<iostream>
+#include<cmath>
+class vec3{
+    public:
+        float x,y,z;
+
+        vec3(){
+            x=y=z=0;
+        }
+
+        //Cartesian coordinate Constructor
+        vec3(float x,float y,float z){
+            this->x=x;
+            this->y=y;
+            this->z=z;
+        }
+        
+        //assignment operator
+        void operator=(vec3 s){
```

Comment:
Usually, the assignment operator returns [a non-const reference to the object that is assigned to](https://stackoverflow.com/a/3106307/12591388).

However, the assignment operator doesn't need to be overloaded in this case. The compiler-generated one is good enough for us.